src/common/hooks/useDialog/index.tsx

```diff
@@ -0,0 +1,38 @@
+import dynamic from 'next/dynamic'
+
+import { useState } from 'react'
+
+import { UniversalDialogProps } from '@/common/components/UniversalDialog/types'
+
+import { UseDialog } from './types'
+
+const DynamicUniversalDialog = dynamic(
+  async () =>
+    (
+      await import(
+        /* webpackChunkName: "UniversalDialog" */
+        '@/common/components/UniversalDialog'
+      )
+    ).UniversalDialog,
+  {
+    ssr: false,
+  }
+)
+
+export const useDialog = (initialProps?: Omit<UniversalDialogProps, 'open' | 'setOpen'>): UseDialog => {
+  const [open, setOpen] = useState(false)
+  const [currentProps, setCurrentProps] = useState<Omit<UniversalDialogProps, 'open' | 'setOpen'>>(initialProps ?? {})
+
+  // TODO better way to handle this?
+  const Dialog = () => <DynamicUniversalDialog {...currentProps} open={open} setOpen={setOpen} />
```

Comment:
Don't do this. You're going to create a new `Dialog` component every time `useDialog` is called.